docs/authentication.md

```diff
@@ -0,0 +1,77 @@
+# Authentication
```

Comment:
Maybe just call this file "Security" since we can talk about encrypted comms, authentication, and later authorization together here? We can append a link like this to `README.md`:

```
* **[Security](security.md)**
  How to secure Helios
```